### ISSUE TITLE
getSelectedRange returns invalid range after row removal

### ISSUE THREAD

**Issue by ottosson** [handsontable/handsontable#7337]
### Description
When selecting all rows/cols/table and remove row/col the current selection range is invalid as the range.to will be larger than whats available.

I don't know if this is the intended behavior, but I would like it to at least check to make sure that range.to is not larger than the table when range.from = -1, i.e. when the whole row/col/table is selected.

### Steps to reproduce

1. Select all
2. Remove row/col
3. Get selected range should now be invalid 

### Demo

https://jsfiddle.net/w4ghju0c/

### Your environment
* Handsontable version: 8.1.0
* Browser Name and version: latest Chrome
* Operating System: Windows 10

**Comment by AMBudnik** [comment]
You're right @ottosson this information is not updated after we remove a row.

We can see that we need to select the cells once again to get the updated value - https://jsfiddle.net/AMBudnik/sbk5td7g/ (printing 'to' selection on a button click). 

I am not sure, though, if that is a bug, cause the selection is not re-triggered. If we do not re-select the range why should we get a new value? I'll keep this issue opened for a discussion.

As a workaround for having the range updated we can call the `selectAll()` once again https://jsfiddle.net/AMBudnik/67jL8s1h/

**Comment by ottosson** [comment]
Yes, selectAll works unless you have multiple ranges, some of which have selected non-existing rows. I have worked around this for now by checking for range.to is not greater than the available row count. Not ideal to have to do that check everywhere if you use that feature a lot. Easy to miss.

**Comment by budnix** [comment]
The issue is already fixed. Demo v17 https://jsfiddle.net/budnix/s67kj4zw/.
